src/adapter/src/catalog.rs

```diff
@@ -584,6 +584,15 @@ impl Catalog {
             .err_into()
     }
 
+    /// Get the next user item ID without allocating it.
+    pub async fn get_next_user_item_id(&self) -> Result<u64, Error> {
```

Comment:
Only bother changing this when pushing a change anyways, but I often find a name like `peek_next_user_item_id` to better conveying the semantics.